stumpy/stump.py

```diff
@@ -162,15 +162,15 @@ def _compute_diagonal(
         else:
             iter_range = range(-k, min(n_A - m + 1, n_B - m + 1 - k))
 
-        for signed_i in iter_range:
-            i = np.uint64(signed_i)
-            j = np.uint64(i + k)
+        for i in iter_range:
+            uint64_i = np.uint64(i)
+            uint64_j = np.uint64(i + k)
 
-            if i == 0 or (k < 0 and i == -k):
+            if uint64_i == 0 or (k < 0 and uint64_i == -k):
```

Comment:
As you said, we are thinking of `i` and `j` as the index of row and column. So, if we are at the first row, or first column, we cannot use a rolling approach to update `cov`, and we need to calculate it directly.

Reply:
Interesting! This makes a lot more logical sense. Are the two lines really equivalent? I can't easily wrap my head around it 😆

Reply:
@NimaSarajpoor agreed, I just think this should be a different issue as it doesn't technically have anything to do with unsigned int optimization. We may also need to search the codebase for similar patterns. What do you think?

Reply:
I leave it up to you @alvii147. If you choose not to do it here, will you please create a new issue pointing to this comment and explaining the relationship?

Reply:
Yeah...I just did bash command `grep -r 'k < 0'` and I only found it in `stump` and `aamp`. However, to be on the safe side, we may want to do a visual check on other modules to make sure we do not have a similar thing in other scripts. So, it MIGHT be worth it to open a  new issue.

---
@seanlaw 
> Are the two lines really equivalent?

I believe so. The case `i == -k` is basically the same as  `i + k == 0`, and as @alvii147 explained before, the condition `k < 0` is not needed since `i` is always non-negative. So, `i + k == 0` itself contains the condition `k < 0`.

Reply:
Alright you've convinced me! I've made this change.